pkg/templates/nstemplatetiers/nstemplatetier_generator.go

```diff
@@ -28,16 +32,33 @@ func CreateOrUpdateResources(ctx context.Context, s *runtime.Scheme, client runt
 	if err != nil {
 		return errors.Wrapf(err, "unable to load templates")
 	}
+
+	paths, err := templates.GetAllFileNames(&nstemplatetierFS, root)
+	if err != nil {
+		return err
+	}
+	if len(paths) == 0 {
+		return fmt.Errorf("could not find any user templates")
+	}
+
 	files := map[string][]byte{}
-	for _, name := range assets.Names() {
-		if name == "metadata.yaml" {
+	for _, name := range paths {
+		if name == "templates/nstemplatetiers/metadata.yaml" {
```

Comment:
This feels tightly coupled to `deploy.NSTemplateTiersFS`. This would be ok if we didn't use the `nstemplatetierFS` as a parameter but IMHO is stylistically wrong when we're allowing passing in an arbitrary FS.

Reply:
updated it